src/ui/public/directives/saved_object_finder.js

```diff
@@ -285,16 +285,17 @@ module.directive('savedObjectFinder', function ($location, $injector, kbnUrl, Pr
         const isLabsEnabled = config.get('visualize:enableLabs');
         self.service.find(filter)
           .then(function (hits) {
-
-            hits.hits = hits.hits.filter((hit) => (isLabsEnabled || _.get(hit, 'type.stage') !== 'lab'));
-            hits.total = hits.hits.length;
-
-            // ensure that we don't display old results
-            // as we can't really cancel requests
-            if (currentFilter === filter) {
-              self.hitCount = hits.total;
-              self.hits = _.sortBy(hits.hits, 'title');
-            }
+            $scope.$apply(() => {
```

Comment:
I think `$evalAsync` is supposed to be better than using `$apply` because you avoid the risk of colliding with an existing digest cycle.